plugins/inputs/beat/beat.go

```diff
@@ -0,0 +1,278 @@
+
+	var method string
+	if beat.Method != "" {
+		method = beat.Method
+	} else {
+		method = "GET"
+	}
+
+	request, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return err
+	}
+
+	if (beat.Username != "") || (beat.Password != "") {
+		request.SetBasicAuth(beat.Username, beat.Password)
+	}
+	for header, value := range beat.Headers {
+		request.Header.Add(header, value)
+	}
+	if beat.HostHeader != "" {
+		request.Host = beat.HostHeader
+	}
+
+	response, err := beat.client.Do(request)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
+
+	err = json.NewDecoder(response.Body).Decode(value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (beat *Beat) gatherInfoTags(url string) (map[string]string, error) {
+	beatInfo := &BeatInfo{}
+
+	err := beat.gatherJsonData(url, beatInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	tags := map[string]string{
+		"beat_beat":    beatInfo.Beat,
+		"beat_id":      beatInfo.UUID,
+		"beat_name":    beatInfo.Name,
+		"beat_host":    beatInfo.Hostname,
+		"beat_version": beatInfo.Version,
+	}
+
+	return tags, nil
+}
+
+func (beat *Beat) gatherStats(accumulator telegraf.Accumulator) error {
+	beatStats := &BeatStats{}
+
+	infoUrl, err := url.Parse(beat.URL + suffixInfo)
+	if err != nil {
+		return err
+	}
+	statsUrl, err := url.Parse(beat.URL + suffixStats)
+	if err != nil {
+		return err
+	}
+
+	tags, err := beat.gatherInfoTags(infoUrl.String())
+	if err != nil {
+		return err
+	}
+
+	err = beat.gatherJsonData(statsUrl.String(), beatStats)
+	if err != nil {
+		return err
+	}
+
+	if beat.CollectBeatStats {
+		flattenerBeat := jsonparser.JSONFlattener{}
+		err := flattenerBeat.FlattenJSON("", beatStats.Beat)
+		if err != nil {
+			return err
+		}
+		accumulator.AddFields("beat", flattenerBeat.Fields, tags)
+	}
+
+	if beat.CollectFilebeatStats {
+		flattenerBeat := jsonparser.JSONFlattener{}
+		err := flattenerBeat.FlattenJSON("", beatStats.FileBeat)
+		if err != nil {
+			return err
+		}
+		accumulator.AddFields("beat_filebeat", flattenerBeat.Fields, tags)
+	}
+
+	if beat.CollectLibbeatStats {
+		flattenerLibbeat := jsonparser.JSONFlattener{}
+		err := flattenerLibbeat.FlattenJSON("", beatStats.Libbeat)
+		if err != nil {
+			return err
+		}
+		accumulator.AddFields("beat_libbeat", flattenerLibbeat.Fields, tags)
+	}
+
+	if beat.CollectSystemStats {
+		flattenerSystem := jsonparser.JSONFlattener{}
+		err := flattenerSystem.FlattenJSON("", beatStats.System)
+		if err != nil {
+			return err
+		}
+		accumulator.AddFields("beat_system", flattenerSystem.Fields, tags)
+	}
+
+	return nil
+}
+
+func (beat *Beat) Gather(accumulator telegraf.Accumulator) error {
+	if beat.client == nil {
+		client, err := beat.createHttpClient()
+
+		if err != nil {
+			return err
+		}
+		beat.client = client
+	}
+
+	err := beat.gatherStats(accumulator)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func init() {
+	inputs.Add("beat", func() telegraf.Input {
+		return NewBeat()
```

Comment:
```suggestion
		&Beat{}
```
and setting the defaults in an `Init()` function.

Reply:
will fix